### PR TITLE
SWC-6049: make file size optional in renderer code

### PR DESCRIPTION
**Pull request by jay-hodgson**
do not supply the byte count if greaterThan is true from the table/view stats response (we do not want to show this minimum to the user)

### CODE REVIEW

**Comment by jay-hodgson** [comment]
Under 100 files, should work the same as it used to:
Screen Shot 2022-04-25 at 11 11 58 AM
But when greaterThan is true (>100 files), only show the file count:
Screen Shot 2022-04-25 at 11 12 23 AM

File: src/lib/containers/download_list/DownloadDetails.tsx [hunk 1]
```diff
@@ -51,28 +52,33 @@ export default function DownloadDetails(props: DownloadDetailsProps) {
   }, [accessToken])
 
   const timeEstimateInSeconds =
-    isLoading || downloadSpeed === 0 ? 0 : numBytes / downloadSpeed
+    isLoading || downloadSpeed === 0 || !numBytes ? 0 : numBytes / downloadSpeed
+  const isTimeEstimateLoading = timeEstimateInSeconds === 0
```
Comment (jay-hodgson) [inline src/lib/containers/download_list/DownloadDetails.tsx:56]: Tried to maintain the same logic that was already in this component ( they seem to use timeEstimateInSeconds=0 to indicate that it should be inactive)

File: src/lib/containers/download_list/DownloadDetails.tsx [hunk 2]
```diff
@@ -51,28 +52,33 @@ export default function DownloadDetails(props: DownloadDetailsProps) {
...
-        {isInactive ? (
+        <FontAwesomeIcon className={fileCountIconClass} icon="file" />
+        {isZeroFiles ? (
           <SkeletonInlineBlock width={50} />
         ) : (
           <> {numFiles}&nbsp;files </>
         )}
       </span>
-      <span
+      {numBytes && <span
```
Comment (jay-hodgson) [inline src/lib/containers/download_list/DownloadDetails.tsx:78]: only show the number of bytes if set (and greater than 0, since 0 is also falsy)

File: src/lib/containers/download_list/DownloadConfirmation.tsx [hunk 3]
```diff
@@ -248,9 +247,10 @@ export const DownloadConfirmation: React.FunctionComponent<DownloadConfirmationP
 
     useDeepCompareEffect(() => {
       if (queryResultBundle && downloadListStatistics) {
+        const isGreaterThanReportedValue = queryResultBundle.sumFileSizes?.greaterThan
```
Comment (jay-hodgson) [inline src/lib/containers/download_list/DownloadConfirmation.tsx:250]: If greaterThan is true, then use an undefined file size sum (indeterminate)

File: src/lib/containers/download_list/DownloadDetails.tsx [hunk 4]
```diff
@@ -31,13 +31,14 @@ type State = {
  * @returns
  */
 export default function DownloadDetails(props: DownloadDetailsProps) {
+  const { numFiles, numBytes } = props
+  const { accessToken } = useSynapseContext()
+
   const [state, setState] = useState<State>({
-    isLoading: true,
+    isLoading: !!numBytes, // loading=true (to figure out the estimated download time) iff we were given a byte count
```
Comment (nickgros) [inline src/lib/containers/download_list/DownloadDetails.tsx:38]: Just dropping old code from the comment, but keeping the useful part
```suggestion
    isLoading: !!numBytes, // figure out the estimated download time iff we were given a byte count
```

**Comment by jay-hodgson** [comment]
Included message change for https://sagebionetworks.jira.com/browse/SWC-6082
Screen Shot 2022-04-25 at 1 41 23 PM

File: src/lib/containers/download_list/DownloadDetails.tsx (hunk 1 above)
Comment (nickgros) [reply src/lib/containers/download_list/DownloadDetails.tsx:56]: Can be hard to follow, but it at least keeps this change simple

**Review by nickgros — APPROVED** [review]
LGTM!
